src/tests/class_tests/openms/source/PeakIntegrator_test.cpp

```diff
@@ -561,22 +589,44 @@ START_SECTION(PeakShapeMetrics calculatePeakShapeMetrics(
   PeakIntegrator::PeakShapeMetrics psm;
   pa = ptr->integratePeak(chromatogram, left, right);
   psm = ptr->calculatePeakShapeMetrics(chromatogram, left, right, pa.height, pa.apex_pos);
-  TEST_REAL_SIMILAR(psm.width_at_5, 0.231263425125414)
-  TEST_REAL_SIMILAR(psm.width_at_10, 0.134762234301732)
-  TEST_REAL_SIMILAR(psm.width_at_50, 0.0595791540757924)
-  TEST_REAL_SIMILAR(psm.start_position_at_5, 2.51268515480125)
-  TEST_REAL_SIMILAR(psm.start_position_at_10, 2.63222565817823)
-  TEST_REAL_SIMILAR(psm.start_position_at_50, 2.65391757114759)
-  TEST_REAL_SIMILAR(psm.end_position_at_5, 2.74394857992666)
-  TEST_REAL_SIMILAR(psm.end_position_at_10, 2.76698789247996)
-  TEST_REAL_SIMILAR(psm.end_position_at_50, 2.71349672522338)
+  TEST_REAL_SIMILAR(psm.width_at_5, 0.15955)
+  TEST_REAL_SIMILAR(psm.width_at_10, 0.138366667)
+  TEST_REAL_SIMILAR(psm.width_at_50, 0.0741500000000004)
+  TEST_REAL_SIMILAR(psm.start_position_at_5, 2.630066667)
+  TEST_REAL_SIMILAR(psm.start_position_at_10, 2.64065)
+  TEST_REAL_SIMILAR(psm.start_position_at_50, 2.662116667)
+  TEST_REAL_SIMILAR(psm.end_position_at_5, 2.789616667)
+  TEST_REAL_SIMILAR(psm.end_position_at_10, 2.779016667)
+  TEST_REAL_SIMILAR(psm.end_position_at_50, 2.736266667)
   TEST_REAL_SIMILAR(psm.total_width, 0.540983333)
-  TEST_REAL_SIMILAR(psm.tailing_factor, 5.86240177860251)
-  TEST_REAL_SIMILAR(psm.asymmetry_factor, 0.864593034054243)
+  TEST_REAL_SIMILAR(psm.tailing_factor, 2.14352889450752)
+  TEST_REAL_SIMILAR(psm.asymmetry_factor, 0.856855285811426)
   TEST_REAL_SIMILAR(psm.slope_of_baseline, 2454)
   TEST_REAL_SIMILAR(psm.baseline_delta_2_height, 0.00253908735640033)
   TEST_EQUAL(psm.points_across_baseline, 57)
   TEST_EQUAL(psm.points_across_half_height, 6)
+  pa = ptr->integratePeak(chromatogram, left_past_5, right_past_5);
```

Comment:
This should cover all of the fringe cases that I have seen so far during our testing with analytics.